stdlib/enum.pyi

```diff
@@ -213,15 +223,21 @@ class Flag(Enum):
     def __and__(self: Self, other: Self) -> Self: ...
     def __xor__(self: Self, other: Self) -> Self: ...
     def __invert__(self: Self) -> Self: ...
+    if sys.version_info >= (3, 11):
+        def __iter__(self: Self) -> Iterator[Self]: ...
+        def __len__(self) -> int: ...
+        __ror__ = __or__
+        __rand__ = __and__
+        __rxor__ = __xor__
 
 class IntFlag(int, Flag):
     def __new__(cls: type[Self], value: int) -> Self: ...
     def __or__(self: Self, other: int) -> Self: ...
     def __and__(self: Self, other: int) -> Self: ...
     def __xor__(self: Self, other: int) -> Self: ...
-    def __ror__(self: Self, other: int) -> Self: ...
-    def __rand__(self: Self, other: int) -> Self: ...
-    def __rxor__(self: Self, other: int) -> Self: ...
+    __ror__ = __or__
+    __rand__ = __and__
+    __rxor__ = __xor__
```

Comment:
This brings the stub closer to the runtime, which uses assignments for these methods. It reverts changes that were made in #6114 by @jakebailey -- it looks like the changes in that PR were made to make pyright happy? Not sure.

Reply:
Hm, I can see now that it was caused by a parameter name clash with a parent class; I'd go check to see if they still differ... It seems weird that we wouldn't report that, unless that rule has been disabled?

Reply:
Ah, I think we might have disabled the relevant check for typeshed: https://github.com/python/typeshed/pull/7498

In typeshed, we generally prefer to match the implementation at runtime, even if the runtime has... dubious respect for LSP in some cases :)

Reply:
This isn't an LSP thing, it's an assignability error because named parameters change how a caller is able to use the function (unless they are marked as positional only). A call site with assuming the names of the parent class will crash on a subclass which changes those names.

The saving grace here is that this is a dunder function that isn't often called directly.

I would hope this rule can be reenabled (and lines like this use an ignore which explicitly mention the rule name or something). It's not great to leave this rule off.

Reply:
> I would hope this rule can be reenabled (and lines like this use an ignore which explicitly mention the rule name or something). It's not great to leave this rule off.

Again, I'm not defending the implementation necessarily! (Though I will note that using method assignment like this is pretty idiomatic Python imo, especially for reflected dunders.) But bugs in the implementation should be reported over at the CPython repo -- our job over here at typeshed is to reflect the runtime. I don't think having lots of `type: ignore`s because of this pyright check particularly helps us to do that :)